### PR TITLE
MNTOR-1632: Port dashboard page

### PR DESCRIPTION
**Pull request by flozia**
# References: 
Jira: [MNTOR-1632](https://mozilla-hub.atlassian.net/browse/MNTOR-1632)

# Description

Ports the dashboard page to Next.js.

# How to test

- Visit `/user/breaches` and/or `/user/dashboard`
- Visit the dashboard page and make sure that the page looks and works as expected

# Checklist (Definition of Done) 
- [x] Localization strings (if needed) have been added.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] I've added or updated the relevant sections in readme and/or code comments
- [x] I've added a unit test to test for potential regressions of this bug.
- [x] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [x] All acceptance criteria are met.
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [x] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: src/app/(nextjs_migration)/(authenticated)/user/layout.tsx
Comment (Vinnl) [inline src/app/(nextjs_migration)/(authenticated)/user/layout.tsx:1]: I can't add a comment to line 25 directly since it didn't technically change in this PR, but if I could add a tiny bit of scope, I think the desired behaviour here is to trigger a sign-in flow when the user is not logged in?

I'm not sure if there's a "good" way to do it, but a dirty way I can think of is to just render a client component that triggers `signIn('fxa')` on render.

File: src/app/(nextjs_migration)/(authenticated)/user/breaches/breaches.d.ts
Comment (Vinnl) [inline src/app/(nextjs_migration)/(authenticated)/user/breaches/breaches.d.ts:1]: Woah, awesome job documenting all these types! Maybe you could also update code that is pointing to `Breach` from `/src/app/transitionTypes.ts` to point to the one in this file?

File: src/app/functions/server/breachResolution.ts
Comment (Vinnl) [inline src/app/functions/server/breachResolution.ts:13]: You could annotate this to conform to the type definition in `breaches.d.ts`, to ensure that we don't forget to update this when we update that.

Alternatively, instead of having that type there, you could replace line 27 with

```ts
} as const;
```

after which you could turn it into a type using:

```ts
export type BreachDataTypesType = typeof BreachDataTypes;
```

(And possibly come up with better names :sweat_smile: )

File: src/app/api/v1/user/breaches/route.ts
Comment (Vinnl) [inline src/app/api/v1/user/breaches/route.ts:1]: :star2: full stack :star2: 

(I assigned the relevant ticket to you - heads up to @mansaj who was also looking into porting API routes.)

**Review by Vinnl — APPROVED** [review]
This is really great work Florian, nice job! By far the biggest chunk of the porting work, and took care of an API route to boot :clap:

File: public/nextjs_migration/client/js/breaches.js
Comment (pdehaan) [inline public/nextjs_migration/client/js/breaches.js:187]: Q: Why do we need `capitalFirstLetter()` aren't the strings already translated in Pontoon?

File: public/nextjs_migration/client/js/circleChart.js
Comment (pdehaan) [inline public/nextjs_migration/client/js/circleChart.js:257]: `font-size` duped here... I'm guessing the first one (the calculated one) is probably ignored. 🤷

File: public/nextjs_migration/client/js/circleChart.js
Comment (pdehaan) [inline public/nextjs_migration/client/js/circleChart.js:282]: Re: `this.chartElement?.`,

Is it possible `chartElement` is missing/`undefined`? Or was that just to appease the linting gods?

File: src/app/api/v1/user/breaches/route.ts
Comment (mansaj) [reply src/app/api/v1/user/breaches/route.ts:1]: Merged https://github.com/mozilla/blurts-server/pull/3083 into this PR, I'll close the other one once this is merged into main

**Review by mansaj — APPROVED** [review]
Given that we will most likely rework these APIs for the new design, approving this for strictly the React work

File: src/app/(nextjs_migration)/(authenticated)/user/breaches/breaches.d.ts
Comment (mansaj) [inline src/app/(nextjs_migration)/(authenticated)/user/breaches/breaches.d.ts:144]: Best to name it something like: `BreachResolutionRequest` because we might have one for the response later.

File: src/app/(nextjs_migration)/(authenticated)/user/breaches/breaches.d.ts
Comment (flozia) [reply src/app/(nextjs_migration)/(authenticated)/user/breaches/breaches.d.ts:1]: Updated! I left `/src/app/transitionTypes.ts` in there so we still have a dedicated place for loose type declarations during porting work: [c2cd69e](https://github.com/mozilla/blurts-server/pull/3085/commits/c2cd69ec08fd2b6621d70d918180c7107a5f2562).

File: src/app/(nextjs_migration)/(authenticated)/user/layout.tsx
Comment (Vinnl) [reply src/app/(nextjs_migration)/(authenticated)/user/layout.tsx:1]: Ah cool, incidentally had to do [the same in my PR](https://github.com/mozilla/blurts-server/pull/3086/files#diff-5bb270ee62cb7e053523891711315de3ed4f0939a981344d9d0e55b9ca66821d), but used a separate component. (I didn't tell you to see if you could come up with a better solution :P ) I'll update to yours after you merge.

(I do think we'll remove `<SignInButton>` at some point in favour of a full user menu, but that shouldn't be a big refactor.)

File: src/app/functions/server/breachResolution.ts
Comment (flozia) [reply src/app/functions/server/breachResolution.ts:13]: That’s a good idea — done in: [5ed06c7](https://github.com/mozilla/blurts-server/pull/3085/commits/5ed06c772ae3772ddd07cbf52bc5ac936c1779d8).

File: public/nextjs_migration/client/js/breaches.js
Comment (flozia) [reply public/nextjs_migration/client/js/breaches.js]: Good point — we should not alter translated strings. Removed: [8f31321](https://github.com/mozilla/blurts-server/pull/3085/commits/8f3132170cdfcb369eafe4f076cdf0b33dbd35b8).

File: public/nextjs_migration/client/js/circleChart.js
Comment (flozia) [reply public/nextjs_migration/client/js/circleChart.js:257]: Yupp — sneaked it’s way in: [b132b0c](https://github.com/mozilla/blurts-server/pull/3085/commits/b132b0c1cca01495aba0da23f82b2723572f0b21).

File: public/nextjs_migration/client/js/circleChart.js
Comment (flozia) [reply public/nextjs_migration/client/js/circleChart.js:282]: I would not expect it to be missing or `undefined` and it really is to make the linter happy.

**Pull request merged**
